src/main.rs

```diff
@@ -25,16 +28,27 @@ enum RustyHookOpts {
     },
 }
 
-fn init() {
+fn init(skip_hook_list: Option<String>) {
     if ci_info::is_ci() {
         println!("[rusty-hook] CI Environment detected. Skipping hook install");
         exit(0);
     }
 
+    let skip_hook_list: Vec<String> = if let Some(skip_hook_list) = skip_hook_list {
+        skip_hook_list
+            .split(",")
+            .map(String::from)
+            .into_iter()
+            .collect()
+    } else {
+        vec![]
+    };
+
```

Comment:
We won't need to represent these as a collection of owned strings in order to figure out the (potential) subset of the full hook lists to lay down, and as you probably noticed doing so causes some minor annoyances later on (such as needing to allocate).

It's certainly true that minor annoyance doesn't have any real tangible run time impact of course, but I think we can just use slices directly and clean things up a bit. For example:

```suggestion
    let skip_hook_list = skip_hook_list
        .as_deref()
        .map_or(vec![], |s| s.split(",").collect());

```

or could even consider skipping the local altogether and just use the above rhs assignment expression as the arg to the `init` function (either way is fine with me!)

Reply:
okay.. i'll take a look into it